azavea/iow-boundary-tool
# Add ability to submit review

## Overview

This PR adds the ability for validators to submit a review. It consists primarily of

- UI components for the finish review process
- API routes/RTK endpoints for finishing a review

Closes #160 

### Demo

https://user-images.githubusercontent.com/8356789/200348357-23593cef-d9e3-4ef6-8e99-900ba755d1a9.mov

### Notes

I split the DrawTools component into separate components for draft and review modes. It started to feel like each had different enough components to warrant this.

## Testing Instructions

- `./scripts/resetdb`
- Login as v1@azavea.com
- Select "Azavea Test Utility"
- Go the the submission that is "In Review"
- Click "Request Changes"
- Enter a comment in the textbox
- Click "Save"
  - [x] Ensure you are redirected to the submission details page
  - [x] Ensure that the status of the submission is "Needs Revisions"
  - [x] Ensure that the review is present in the activity log
  - [x] Ensure the author, time, and comment are accurate

## Checklist
